flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/declarative/StateWithExecutionGraph.java

```diff
@@ -106,14 +107,17 @@
                                 context.getMainThreadExecutor()));
     }
 
+    @VisibleForTesting
```

Comment:
Why did you add this annotation? Isn't this method also used by the states itself

Reply:
Since the scheduler should not be able to access them, I assumed this was done for testing purposes.
If they were simply made accessible to sub-classes, then they should be made protected to convey this.